### PR TITLE
tools: Convert docuploader and release-progress-reporter into dotnet tools

### PR DESCRIPTION
**Pull request by jskeet**
We won't want to merge this until we've successfully released libraries with the previous approach.

The first commit is from #10956.

### CODE REVIEW

File: tools/Google.Cloud.Tools.DocUploader/Google.Cloud.Tools.DocUploader.csproj [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
+  <Import Project="../DotnetToolProperties.xml" />
+
   <PropertyGroup>
     <OutputType>Exe</OutputType>
     <TargetFramework>net6.0</TargetFramework>
     <ImplicitUsings>enable</ImplicitUsings>
     <Nullable>enable</Nullable>
+
+    <Version>0.1.0</Version>
+    <Description>Tool used by Google .NET release processes for documentation. While there is nothing "secret" in this package, it is unlikely to be useful to other developers.</Description>
```
Comment (amanda-tarafa) [inline tools/Google.Cloud.Tools.DocUploader/Google.Cloud.Tools.DocUploader.csproj:12]: nit: Maybe adding a note saying, "we just publish this for convenience" to make emphasis. Here and in the other one.

**Review by amanda-tarafa — APPROVED** [review]
LGTM.

As for the exception, I think it's fine to use the plain `Exception`, else linking the code file seems like the best alternative.

File: tools/Google.Cloud.Tools.DocUploader/Google.Cloud.Tools.DocUploader.csproj (hunk 1 above)
Comment (jskeet) [reply tools/Google.Cloud.Tools.DocUploader/Google.Cloud.Tools.DocUploader.csproj:12]: Yup, will do.

File: tools/Google.Cloud.Tools.ReleaseProgressReporter/Program.cs [hunk 2]
```diff
@@ -12,14 +12,13 @@
...
 
-using Google.Cloud.Tools.Common;
 using Google.Cloud.Tools.ReleaseProgressReporter;
 using Octokit;
 
 // Note: CommandLineParser has not been used here for the sake of simplicity.
 if (args.Length < 1)
 {
-    throw new UserErrorException("Please specify the progress point: 'start' or 'finish'.");
+    throw new Exception("Please specify the progress point: 'publish-reporter-script', 'start' or 'finish'.");
```
Comment (jskeet) [inline tools/Google.Cloud.Tools.ReleaseProgressReporter/Program.cs:21]: This message isn't actually accurate just yet, but it is in a later commit :)

**Pull request merged**